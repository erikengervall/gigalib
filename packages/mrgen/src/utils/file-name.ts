import path from 'node:path';

export function fileName(file: string) {
  const { name, base } = path.parse(file);

  return {
    withExt: base,
    withoutExt: name,
  };
}
