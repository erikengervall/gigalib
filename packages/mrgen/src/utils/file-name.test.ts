import { fileName } from './file-name';

it('should return file name withExt and withoutExt', () => {
  expect(fileName('test.txt')).toMatchSnapshot();
  expect(fileName('a/b/c/test.txt')).toMatchSnapshot();
  expect(fileName('/a/b/c/test.txt')).toMatchSnapshot();
  expect(fileName('test')).toMatchSnapshot();
});
