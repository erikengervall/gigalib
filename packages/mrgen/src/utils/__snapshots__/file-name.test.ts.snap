// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return file name withExt and withoutExt 1`] = `
{
  "withExt": "test.txt",
  "withoutExt": "test",
}
`;

exports[`should return file name withExt and withoutExt 2`] = `
{
  "withExt": "test.txt",
  "withoutExt": "test",
}
`;

exports[`should return file name withExt and withoutExt 3`] = `
{
  "withExt": "test.txt",
  "withoutExt": "test",
}
`;

exports[`should return file name withExt and withoutExt 4`] = `
{
  "withExt": "test",
  "withoutExt": "test",
}
`;
